--- 
title: "scMINER: a mutual information-based framework for identifying hidden drivers from single-cell omics data"
author: "Qingfei Pan"
date: "`r Sys.Date()`"
knit: "bookdown::render_book"
site: "bookdown::bookdown_site"
documentclass: "krantz"
description: "A tutorial of scMINER"
biblio-style: "apalike"
link-citations: yes
colorlinks: yes
lot: yes
lof: yes
fontsize: 12pt
monofont: "Source Code Pro"
monofontoptions: "Scale=0.7"
github-repo: QingfeiPan/scMINER_qpan
---


# <img src="images/scMINER_logo.png" alt="" width="250px"> {-}

---

Welcome to scMINER documentation!

**scMINER** (**s**ingle-**c**ell **M**utual **I**nformation-based **N**etwork **E**ngineering **R**anger) is a computational framework designed for **end-to-end** analysis of single cell RNA-seq data. Using [**mutual information**](https://en.wikipedia.org/wiki/Mutual_information) to measure cell-cell similarities and gene-gene correlations, scMINER is widely applicable and highly accurate in unsupervised clustering and gene activity inference of scRNA-seq data.

In this documentation, we will walk you through every analysis that scMINER can do and introduce you more about the concepts related to scMINER framework.

<center><img src="images/scMINER_overview.jpg" alt="centered image" width="700px"></center>

<!--chapter:end:index.Rmd-->

# Introduction

This chapter will introduce 

## A few concepts

There are a few concepts that may help you understand scMINER better.

### SparseEset {-}

### Mutual Information {-}

### Gene Activity {-}


## Why use scMINER

(more deteils to be added)

scMINER includes the following key functions:

- **Mutual information-based clustering**: scMINER measures the cell-cell similarities with **full feature-derived mutual information**. It can catch both linear and non-linear correlations and performs better in cell clustering, especially for those of close states.

- **Gene activity estimation**: scMINER rewires the cell-type specific gene networks solely from the scRNA-seq data, and then estimates the gene activities of not only **transcription factors (TFs)** but also **signaling genes (SIGs)**. The gene activity-based analysis can expose the **main regulators of various biological activities**, like cellular linage differentiation and tissue specificity.

- **SparseEset-centered full-feature tool**: scMINER provides a wide range of functions for **data intake**, **quality control and filtration**, **MI-based clustering**, **network inference**, **gene activity estimation**, **cell type annotation**, **differential expression/activity analysis**, and **data visualization and sharing**. Most of these functions are developed in an object-oriented manner for the **SparseEset object**.


## Citation

Please consider citing [this paper](https://www.biorxiv.org/content/10.1101/2023.01.26.523391v1) if you find scMINER useful in your research.


## Support
We welcome your feedback! The scMINER software is developed and maintained by the [Yu Lab @ St. Jude Children's Research Hospital](https://www.stjude.org/research/labs/yu-lab.html) and is released under the Apache License (Version 2.0). Feel free to open an [issue](https://github.com/jyyulab/scMINER/issues), or send us an [email](mailto:jiyang.yu@stjude.org) if you encounter a bug, need our help or just want to make a comment/suggestion.


<!--chapter:end:01-introduction.Rmd-->

# Get started

This chapter will walk you through how to install scMINER, create a project space for your study and prepare the demo data used in this tutorial.


## Installation

scMINER framework is mainly developed with R for its advantages in statistical analysis and data visualization. It also includes two components, [MICA](https://github.com/jyyulab/MICA) and [SJARACNe](https://github.com/jyyulab/SJARACNe), that are developed with Python to take its strengths in calculation speed and memory consumption, since mutual information estimation of large-scale scRNA-seq data is usually compute-intensive.

Please install all three components for the full access to scMINER framework.

### Install scMINER R package {-}

The scMINER R package requires R 4.2.3 or newer, and can be installed from [GitHub](https://github.com/) with:
``` r
# install.packages("devtools")
devtools::install_github("https://github.com/jyyulab/scMINER.git@dev-qpan")
```

### Install MICA and SJARACNe {-}

The recommended method to install MICA and SJARACNe is to use [conda](https://conda.io/projects/conda/en/latest/) dependency manager:

``` bash
## setup conda env
conda create -n scminer python=3.9.2                # Create a python virtual environment
source activate scminer                             # Activate the virtual environment

## install MICA
git clone https://github.com/jyyulab/MICA           # Clone the MICA repo
cd MICA                                             # Switch to the MICA root directory
pip install .                                       # Install MICA and its dependencies
mica -h                                             # Check if MICA works

## install SJARACNE
cd ..                                               # Switch to conda env folder
git clone https://github.com/jyyulab/SJARACNe.git   # Clone the SJARACNe repo
cd SJARACNe                                         # Switch to the MICA root directory
python setup.py build                               # Build SJARACNe binary
python setup.py install                             # Build SJARACNe binary
sjaracne -h                                         # Check if SJARACNe works
```


## Demo data

In this tutorial, we will use a ***ground truth*** dataset called **PBMC14k** for demonstration purposes. It was generated from a Peripheral Blood Mononuclear Cells (PBMCs) dataset containing 10 known cell types, with 2,000 cells per type [[Zheng et al., 2017](./data-reference.md#[Zheng et al., 2017])]:

1. We first rectified the gene symbol issues of the original dataset, including the dash-dot conversion (e.g. “RP11-34P13.7” changed to “RP11.34P13.7”) and “X” added to those started with numbers (e.g. “7SK” changed to “X7SK”), by referring to the gene annotation file (GRCh37.82) used in the original study.
2. Then we removed 3 cell populations, CD34+ cells, CD4+ helper T cells, and total CD8+ cytotoxic cells, from the dataset because of either low sorting purity or a significant overlap with other immune cells based on the sorting strategy, and created a new dataset with seven known cell types and 14k cells in total.

The original dataset is freely available under this accession number [SRP073767](https://www.ncbi.nlm.nih.gov/sra?term=SRP073767) and [Zenodo](https://zenodo.org/record/3357167#.YhQNF2RKj6V).

<details>
<summary>**How was the PBMC14K dataset generated from the original dataset?**</summary>
```{r, demo-dataset-preparation, eval=FALSE}
## Step 1: rectify the invalid gene symbols
# "Filtered_DownSampled_SortedPBMC_data.csv" is the raw count matrix directly downloaded from Zenodo
counts <- read.csv("Filtered_DownSampled_SortedPBMC_data.csv", row.names = 1) 
d <- t(counts); dim(d) # it includes 21592 genes and 20000 cells

# "genesymbol_from_GTF_GRCh37.txt" contains the official gene ids and symbols extracted from GTF file downloaded from
officialGene <- read.table("genesymbol_from_GTF_GRCh37.txt", header = T, sep = "\t", quote = "", stringsAsFactors = F); head(officialGene)  https://ftp.ensembl.org/pub/grch37/current/gtf/homo_sapiens/
officialGene$dotted_symbol <- gsub("-", "\\.", officialGene$gene_name); officialGene$dotted_symbol <- make.unique(officialGene$dotted_symbol)
table(row.names(d) %in% officialGene$dotted_symbol); row.names(d)[! row.names(d) %in% officialGene$dotted_symbol] # two genes are not in: X7SK.1 and X7SK.2
row.names(d) <- gsub("X7SK.1", "7SK", row.names(d)); row.names(d) <- gsub("X7SK.2", "7SK.1", row.names(d))
table(row.names(d) %in% officialGene$dotted_symbol) # all true
row.names(officialGene) <- officialGene$dotted_symbol
officialGene <- officialGene[row.names(d),]
row.names(d) <- make.unique(officialGene$gene_name)

# "Labels.csv" contains the true labels of cell types and was directly downloaded from Zenodo
celltype <- read.csv("Labels.csv"); head(celltype); 
table(celltype$x) # 2000 cells for each of 10 cell types: CD14+ Monocyte, CD19+ B, CD34+, CD4+ T Helper2, CD4+/CD25 T Reg, CD4+/CD45RA+/CD25- Naive T, CD4+/CD45RO+ Memory, CD56+ NK, CD8+ Cytotoxic T, CD8+/CD45RA+ Naive Cytotoxic
df <- data.frame(cell_barcode = colnames(d), trueLabel_full = celltype$x); dim(df)
truelabel_map <- c(`CD14+ Monocyte`="Monocyte", `CD19+ B`="B", `CD34+`="CD34pos", `CD4+ T Helper2`="CD4Th2", `CD4+/CD25 T Reg`="CD4Treg",
                  `CD4+/CD45RA+/CD25- Naive T`="CD4TN", `CD4+/CD45RO+ Memory`="CD4TCM", `CD56+ NK`="NK", `CD8+ Cytotoxic T`="CD8CTL", `CD8+/CD45RA+ Naive Cytotoxic`="CD8TN")
df$trueLabel <- as.character(truelabel_map[df$trueLabel_full])

## Step 2: extract 7 populations
df.14k <- df[df$trueLabel_full %in% c("CD14+ Monocyte", "CD19+ B", "CD4+/CD25 T Reg", "CD4+/CD45RA+/CD25- Naive T", "CD4+/CD45RO+ Memory", "CD56+ NK", "CD8+/CD45RA+ Naive Cytotoxic"),]
write.table(df.14k, file = "PBMC14k_trueLabel.txt", sep = "\t", row.names = TRUE, col.names = TRUE, quote = FALSE, append = FALSE)

d.14k <- d[,df.14k$cell_barcode]
d.14k <- d.14k[rowSums(d.14k) > 0,]
write.table(d.14k, file = "PBMC14k_rawCount.txt", sep = "\t", row.names = FALSE, col.names = TRUE, quote = FALSE, append = FALSE) # 17986 genes, 14000 cells
```
</details>
\

The **PBMC14k** dataset is embeded in scMINER R package and can be easily loaded by:
```{r check-pbmc14k-dataset, message=FALSE}
library(scMINER)
data("pbmc14k_rawCount")
dim(pbmc14k_rawCount)
pbmc14k_rawCount[1:5,1:5]
```

## Create project space

The project space created by scMINER is a folder of specified name in specified directory. It contains 4 subfolders:

- **`DATA`**: to save the sparse eSet objects and other files;
- **`MICA`**: to save the inputs and outputs of mutual information-based clustering analysis;
- **`SJARACNe`**: to save the inputs and outputs of network inference and quality control;
- **`PLOT`**: to save the files of data visualization.

The project space can not only keep your data centralized and organized, but also make the scMINER pipeline more smooth and robust. We strongly recommend you to create a project space for each of your studies.

This can be easily done by `createProjectSpace()` in scMINER:
```{r create-project-space, eval=FALSE}
scminer_dir <- createProjectSpace(project_dir = "/your-path", project_name = "PBMC14k")
```

The command above will create a folder named **`PBMC14k`** in **`/your-path`**, and save the path to the project space (**`/your-path/PBMC14k`**) to `scminer_dir`.

<details>
<summary>**Can I add, delete or modify files in project space folder?**</summary>
Yes, you can.

- There are two functions, `drawNetworkQC()` and `getActivityBatch()`, that take directories as inputs, and both of them can validate the inputs. For all the rest functions, the inputs are specific files. So adding files in project space **never** affect the scMINER analysis.
- Deleting or modifying files in project spare is also safe. The input validation features of scMINER functions can help locate the files with issues. All output files of scMINER are reproducible and can be re-generated quickly. Just be careful with the clustering results in **`MICA`** and network files in **`SJARACNe`**, ad regerating them can take some time. 
</details>
\

<!--chapter:end:02-get-started.Rmd-->

# Data intake

In this chapter, we will introduce you how to create the SparseExpressionSet(SparseEset) objects from inputs of multiple formats.

## Genereate gene expression matrix

scMINER provides portable functions to generate the **gene expression matrix** from 4 most common file formats of scRNA-seq data, including **sparse matrix by 10x Genomics**, **text-table file**, **HDF5 file by 10x Genomics** and **H5AD file**.

### From data directory by 10x Genomics

This is the most popular input format of scRNA-seq data generated by 10x Genomics. Usually, the data directory contains three files:

- **matrix.mtx**: a sparse matrix format containing the raw UMI count per cell-gene combination
- **barcodes.tsv**: a tab-separated matrix containing the cell barcodes
- **features.tsv**: a tab-separated matrix containing the features/genes and their annotations

For more details about this format, please check out [here](https://www.10xgenomics.com/support/software/cell-ranger/latest/analysis/outputs/cr-outputs-mex-matrices).

```{r read-10x-data-from-directory}
data_dir <- system.file("extdata/demo_inputs/cell_matrix_10x", package = "scMINER")
list.files(path = data_dir, full.names = FALSE)
demo1_mtx <- readInput_10x.dir(input_dir = data_dir, featureType = "gene_symbol", removeSuffix = TRUE, addPrefix = "demo1")
demo1_mtx[1:5,1:5]
```

The `readInput_10x.dir()` function can handle these conditions:

- Alternative file names for feature data: for the datasets generated by CellRanger 3.0 or earlier, the file name is **genes.tsv**;
- Compressed input files: one or more input files are compressed, usually in "**.gz**" format;
- Data with multiple modalities: like the single cell multiome data. In this case, `readInput_10x.dir()` only retains the data of "Gene Expression" by default.


### From text-table file

This is definitely the most compatible text format for scRNA-seq data. which can be used by all single-cell RNA-seq technologies, like 10x Genomics, Smart-Seq, Drop-Seq and more. The commonly used text table file formats include **txt** (text file format), **csv** (comma-separated values) and **tsv** (tab-separated values).

```{r read-10x-data-from-table-file, eval=FALSE}
table_file <- system.file("extdata/demo_inputs/table_file/demoData3.txt.gz", package = "scMINER")
demo3_mtx <- readInput_table(table_file = table_file, sep = "\t", is.geneBYcell = TRUE, removeSuffix = TRUE, addPrefix = "demo3") # set is.geneBYcell = FALSE to read features in columns and cell in in rows
```

***NOTE:*** A major concern to read the gene expression matrix from text-table files is that **the special characters in column names might change to dots (".")**, especially when the matrix is organized in cells by genes. This may cause failures in the identification of mitochondrial genes (usually defined by "MT-|mt-") or spike-in RNAs (usually defined by "ERCC-|Ercc-"). The `readInput_table()` function has set `check.names = FALSE` to avoid this issue. However, if this issue already exists in the source data, you will have to fix it manually.

### From HDF5 file by 10x Genomics

This is another popular input format of scRNA-seq data generated by 10x Genomics. The Hierarchical Data Format version 5 (HDF5 or H5) is a binary format that can compress and access data much more efficiently than text formats. It's super useful when dealing with large datasets.

For more details about this format, please check out [here](https://www.10xgenomics.com/support/software/cell-ranger/latest/analysis/outputs/cr-outputs-h5-matrices).

```{r read-10x-data-from-h5, eval=FALSE}
h5_file <- system.file("extdata/demo_inputs/hdf5_10x/demoData2.h5", package = "scMINER")
demo2_mtx <- readInput_10x.h5(h5_file = h5_file, featureType = "gene_symbol", removeSuffix = TRUE, addPrefix = "demo2")
```

***NOTE:*** The `readInput_10x.h5()` function is developed exclusively for the HDF5 file generated by CellRanger of 10x Genomics. The HDF5 files from other source may have different hierarchical structures and can not be read by this function.


### From H5AD file

The H5AD file is another well-used format for scRNA-seq data. Derived from HDF5 file format, the H5AD file is designed to store large amounts of data efficiently and to facilitate fast access to subsets of the data. Now it's getting more and more popular in scRNA-seq data analysis, visualization and sharing.

For more details about this format, please check out [here](https://anndata.dynverse.org/).

```{r read-10x-data-from-h5ad-file, eval=FALSE}
h5ad_file <- system.file("extdata/demo_inputs/h5ad_file/demoData4.h5ad", package = "scMINER")
demo4_obj <- readInput_h5ad(h5ad_file = h5ad_file, removeSuffix = TRUE, addPrefix = "demo4") # set is.geneBYcell = FALSE to read features in columns and cell in in rows
```

***NOTE:*** Unlike the other three readInput functions which return a gene expression matrx, the `readInput_h5ad()` returns an AnnData object. Here are the key components of an AnnData object:

- **X**: the primary data matrix, cells by genes;
- **obs**: observations (cells) metadata;
- **var**: variables (features/genes) metadata.

## Create SparseEset object

### Solely from the gene expression matrix

This is **the most commonly used way** to create the sparse eSet object with scMINER:
```{r create-sparse-eset-solely-from-matrix}
pbmc14k_raw.eset <- createSparseEset(input_matrix = pbmc14k_rawCount, projectID = "PBMC14k", addMetaData = TRUE)

pbmc14k_raw.eset
```
- **input_matrix**: it's usually but not limited to a sparse matrix of raw UMI count.
  - As for the data format, it accepts **`dgCMatrix`**, **`dgTMatrix`**, **`dgeMatrix`**, **`matrix`**, **`data.frame`**.
  - As for the type of quantification measures, it takes **raw counts**, normalized counts (e.g. **`CPM`** or **`CP10k`**), **`TPM`** (Transcripts Per Million), **`FPKM/RPKM`** (Fragments/Reads Per Kilobase of transcript per Million) and others.
  - **What if a data frame object is given to it?** When a non-matrix table is passed to `input_matrix` argument, the `createSparseEset()` function will automatically convert it to a matrix. And it the matrix, either converted from other format or directly passed from users, is not sparse. `createSparseEset()` will automatically convert it into sparse matrix, by default. This is controlled by another argument called `do.sparseConversion`, the default of which is `TRUE`. It's not recommended but the users can set it as `FALSE` to disable the conversion. Then `createSparseEset()` will create the eSet based on the regular matrix.
- **addMetaData**: when this argument is set `TRUE` (this is the default), `createSparseEset()` will automatically generate 5 statistics, 4 for cells and 1 for features, and add them into the `phenoData` and `featureData` slots. These 5 statistics will be used in quality control and data filtration.

```{r sparse-eset-metadata}
## check the phenoData: metadata of cells
head(pData(pbmc14k_raw.eset))

## check the featureData: metadata of features
head(fData(pbmc14k_raw.eset))
```

### Using self-customized meta data

In some cases, you may have more meta data of either cells (e.g. sample id, treatment condition) or features (e.g. gene full name, gene type, genome location) which will be used in downstream analysis and you do want to add them into the sparse eSet object. the `createSparseEset()` function provides another two arguments, **`cellData`** and **`featureData`**, to take the self-customized meta data. For the PBMC14k dataset, we have the true labels of cell type and would like to add them to the sparse eSet object.

```{r create-sparse-eset-solely-with-cutomized-matadata}
## read the true labels of cell type for PBMC14k dataset
true_label <- read.table(system.file("extdata/demo_pbmc14k/PBMC14k_trueLabel.txt.gz", package = "scMINER"), header = T, row.names = 1, sep = "\t", quote = "", stringsAsFactors = FALSE)

head(true_label)

table(true_label$trueLabel_full)

## the true_label much cover all cells in the expression matrix
table(colnames(pbmc14k_rawCount) %in% row.names(true_label))

## create the sparse eSet object using the true_label
pbmc14k_raw.eset <- createSparseEset(input_matrix = pbmc14k_rawCount, cellData = true_label, featureData = NULL, projectID = "PBMC14k", addMetaData = TRUE)

## check the true labels of cell type from sparse eSet object
head(pData(pbmc14k_raw.eset))
table(pData(pbmc14k_raw.eset)$trueLabel_full)
```

## #From multiple samples

**What if you have multiple samples for one project?** Now it's pretty common to profile multiple samples of the same tissue but under different conditions (e.g. drug treatment) in one project. Analyzing these samples one by one is crucial, and analyzing them in a combined manner may give you more prospects. For this purpose, scMINER provides a function, `combineSparseEset()`, to easily combine the sparse eSet objects of multiple samples.

```{r combine-sparse-eset-readinput, echo=FALSE}
## create a sparse eSet object of each sample to combined
demo1_mtx <- readInput_10x.dir(input_dir = system.file("extdata/demo_inputs/cell_matrix_10x", package = "scMINER"), featureType = "gene_symbol", removeSuffix = TRUE)
demo1.eset <- createSparseEset(input_matrix = demo1_mtx, projectID = "demo1", addMetaData = TRUE)

demo2_mtx <- readInput_table(table_file = system.file("extdata/demo_inputs/table_file/demoData3.txt.gz", package = "scMINER"), sep = "\t", is.geneBYcell = TRUE, removeSuffix = TRUE)
demo2.eset <- createSparseEset(input_matrix = demo2_mtx, projectID = "demo2", addMetaData = TRUE)
```

```{r combine-sparse-eset-merge}
## combine the 4 sparse eSet objects
combined.eset <- combineSparseEset(eset_list = c(demo1.eset, demo2.eset),
                                   projectID = c("sample1", "sample2"),
                                   addPrefix = c("demo1", "demo2"),
                                   addSurfix = NULL, addMetaData = TRUE, imputeNA = TRUE)
dim(combined.eset)
```

A few questions you may have about the `combineSparseEset()` function:
- **What if the input eSets have different features?** `combineSparseEset()` ALWAYS keep all features from the input eSets, and generate NA values wherever the data is not available. By default, this function impute the NA values with the minimum value of the combined matrix, which is usually but not always zero. If this imputation method doesn't fit your study, you can set `imputeNA` to `FALSE` to disable it. If so, the NAs will retain in the eSet object, and you can manually impute them with your own method.
- **What if the input eSets have some same cell barcodes?** `combineSparseEset()` ALWAYS keep all cells from the input eSets, and will report an error when same barcodes are found in different input eSets. This function provides two arguments, `addPrefix` and `addSurfix`, to solve this issue. You can easily avoid the same barcodes of different input eSets by adding a eSet-specific prefix and/or surfix to the barcodes.
```{r show-barcodes-of-combined-eset}
head(pData(combined.eset))
```

- **I have some customized column in the phenoData and/or featureData slots. How does `combineSparseEset()` handle them?** `combineSparseEset()` only keep the columns of phenoData and featureData that shared by all input eSets. Your customized columns would be kept only when they are available in all input eSets.
- **Are the 5 meta data statistics in the combined eSet still same with those generated in each eSet?** No. By default, `combineSparseEset()` will update (add, if they are not available in input eSets) these 5 meta data statistics based on the combined matrix. It's not recommended but you can disable it by setting `addMataData` to `FALSE`.



<!--chapter:end:03-data-intake.Rmd-->

# Footnotes and citations 

## Footnotes

Footnotes are put inside the square brackets after a caret `^[]`. Like this one ^[This is a footnote.]. 

## Citations

Reference items in your bibliography file(s) using `@key`.

For example, we are using the **bookdown** package [@R-bookdown] (check out the last code chunk in index.Rmd to see how this citation key was added) in this sample book, which was built on top of R Markdown and **knitr** [@xie2015] (this citation was added manually in an external file book.bib). 
Note that the `.bib` files need to be listed in the index.Rmd with the YAML `bibliography` key.


The RStudio Visual Markdown Editor can also make it easier to insert citations: <https://rstudio.github.io/visual-markdown-editing/#/citations>

<!--chapter:end:04-citations.Rmd-->

# Blocks

## Equations

Here is an equation.

\begin{equation} 
  f\left(k\right) = \binom{n}{k} p^k\left(1-p\right)^{n-k}
  (\#eq:binom)
\end{equation} 

You may refer to using `\@ref(eq:binom)`, like see Equation \@ref(eq:binom).


## Theorems and proofs

Labeled theorems can be referenced in text using `\@ref(thm:tri)`, for example, check out this smart theorem \@ref(thm:tri).

::: {.theorem #tri}
For a right triangle, if $c$ denotes the *length* of the hypotenuse
and $a$ and $b$ denote the lengths of the **other** two sides, we have
$$a^2 + b^2 = c^2$$
:::

Read more here <https://bookdown.org/yihui/bookdown/markdown-extensions-by-bookdown.html>.

## Callout blocks


The R Markdown Cookbook provides more help on how to use custom blocks to design your own callouts: https://bookdown.org/yihui/rmarkdown-cookbook/custom-blocks.html

<!--chapter:end:05-blocks.Rmd-->

# Sharing your book

## Publishing

HTML books can be published online, see: https://bookdown.org/yihui/bookdown/publishing.html

## 404 pages

By default, users will be directed to a 404 page if they try to access a webpage that cannot be found. If you'd like to customize your 404 page instead of using the default, you may add either a `_404.Rmd` or `_404.md` file to your project root and use code and/or Markdown syntax.

## Metadata for sharing

Bookdown HTML books will provide HTML metadata for social sharing on platforms like Twitter, Facebook, and LinkedIn, using information you provide in the `index.Rmd` YAML. To setup, set the `url` for your book and the path to your `cover-image` file. Your book's `title` and `description` are also used.



This `gitbook` uses the same social sharing data across all chapters in your book- all links shared will look the same.

Specify your book's source repository on GitHub using the `edit` key under the configuration options in the `_output.yml` file, which allows users to suggest an edit by linking to a chapter's source file. 

Read more about the features of this output format here:

https://pkgs.rstudio.com/bookdown/reference/gitbook.html

Or use:

```{r eval=FALSE}
?bookdown::gitbook
```



<!--chapter:end:06-share.Rmd-->

`r if (knitr::is_html_output()) '
# References {-}
'`

<!--chapter:end:07-references.Rmd-->

