<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 3 Data intake | scMINER: a mutual information-based framework for identifying hidden drivers from single-cell omics data</title>
  <meta name="description" content="A tutorial of scMINER" />
  <meta name="generator" content="bookdown 0.40 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 3 Data intake | scMINER: a mutual information-based framework for identifying hidden drivers from single-cell omics data" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="A tutorial of scMINER" />
  <meta name="github-repo" content="QingfeiPan/scMINER_qpan" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 3 Data intake | scMINER: a mutual information-based framework for identifying hidden drivers from single-cell omics data" />
  
  <meta name="twitter:description" content="A tutorial of scMINER" />
  

<meta name="author" content="Qingfei Pan" />


<meta name="date" content="2024-07-29" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="get-started.html"/>
<link rel="next" href="footnotes-and-citations.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A tutorial of scMINER</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><img src="images/scMINER_logo.png" alt="" width="250px"></a></li>
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="introduction.html"><a href="introduction.html#a-few-concepts"><i class="fa fa-check"></i><b>1.1</b> A few concepts</a>
<ul>
<li class="chapter" data-level="" data-path="introduction.html"><a href="introduction.html#sparseeset"><i class="fa fa-check"></i>SparseEset</a></li>
<li class="chapter" data-level="" data-path="introduction.html"><a href="introduction.html#mutual-information"><i class="fa fa-check"></i>Mutual Information</a></li>
<li class="chapter" data-level="" data-path="introduction.html"><a href="introduction.html#gene-activity"><i class="fa fa-check"></i>Gene Activity</a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="introduction.html"><a href="introduction.html#why-use-scminer"><i class="fa fa-check"></i><b>1.2</b> Why use scMINER</a></li>
<li class="chapter" data-level="1.3" data-path="introduction.html"><a href="introduction.html#citation"><i class="fa fa-check"></i><b>1.3</b> Citation</a></li>
<li class="chapter" data-level="1.4" data-path="introduction.html"><a href="introduction.html#support"><i class="fa fa-check"></i><b>1.4</b> Support</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="get-started.html"><a href="get-started.html"><i class="fa fa-check"></i><b>2</b> Get started</a>
<ul>
<li class="chapter" data-level="2.1" data-path="get-started.html"><a href="get-started.html#installation"><i class="fa fa-check"></i><b>2.1</b> Installation</a>
<ul>
<li class="chapter" data-level="" data-path="get-started.html"><a href="get-started.html#install-scminer-r-package"><i class="fa fa-check"></i>Install scMINER R package</a></li>
<li class="chapter" data-level="" data-path="get-started.html"><a href="get-started.html#install-mica-and-sjaracne"><i class="fa fa-check"></i>Install MICA and SJARACNe</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="get-started.html"><a href="get-started.html#demo-data"><i class="fa fa-check"></i><b>2.2</b> Demo data</a></li>
<li class="chapter" data-level="2.3" data-path="get-started.html"><a href="get-started.html#create-project-space"><i class="fa fa-check"></i><b>2.3</b> Create project space</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="data-intake.html"><a href="data-intake.html"><i class="fa fa-check"></i><b>3</b> Data intake</a>
<ul>
<li class="chapter" data-level="3.1" data-path="data-intake.html"><a href="data-intake.html#genereate-gene-expression-matrix"><i class="fa fa-check"></i><b>3.1</b> Genereate gene expression matrix</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="data-intake.html"><a href="data-intake.html#from-data-directory-by-10x-genomics"><i class="fa fa-check"></i><b>3.1.1</b> From data directory by 10x Genomics</a></li>
<li class="chapter" data-level="3.1.2" data-path="data-intake.html"><a href="data-intake.html#from-text-table-file"><i class="fa fa-check"></i><b>3.1.2</b> From text-table file</a></li>
<li class="chapter" data-level="3.1.3" data-path="data-intake.html"><a href="data-intake.html#from-hdf5-file-by-10x-genomics"><i class="fa fa-check"></i><b>3.1.3</b> From HDF5 file by 10x Genomics</a></li>
<li class="chapter" data-level="3.1.4" data-path="data-intake.html"><a href="data-intake.html#from-h5ad-file"><i class="fa fa-check"></i><b>3.1.4</b> From H5AD file</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="data-intake.html"><a href="data-intake.html#create-sparseeset-object"><i class="fa fa-check"></i><b>3.2</b> Create SparseEset object</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="data-intake.html"><a href="data-intake.html#solely-from-the-gene-expression-matrix"><i class="fa fa-check"></i><b>3.2.1</b> Solely from the gene expression matrix</a></li>
<li class="chapter" data-level="3.2.2" data-path="data-intake.html"><a href="data-intake.html#using-self-customized-meta-data"><i class="fa fa-check"></i><b>3.2.2</b> Using self-customized meta data</a></li>
<li class="chapter" data-level="3.2.3" data-path="data-intake.html"><a href="data-intake.html#from-multiple-samples"><i class="fa fa-check"></i><b>3.2.3</b> From multiple samples</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="footnotes-and-citations.html"><a href="footnotes-and-citations.html"><i class="fa fa-check"></i><b>4</b> Footnotes and citations</a>
<ul>
<li class="chapter" data-level="4.1" data-path="footnotes-and-citations.html"><a href="footnotes-and-citations.html#footnotes"><i class="fa fa-check"></i><b>4.1</b> Footnotes</a></li>
<li class="chapter" data-level="4.2" data-path="footnotes-and-citations.html"><a href="footnotes-and-citations.html#citations"><i class="fa fa-check"></i><b>4.2</b> Citations</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="blocks.html"><a href="blocks.html"><i class="fa fa-check"></i><b>5</b> Blocks</a>
<ul>
<li class="chapter" data-level="5.1" data-path="blocks.html"><a href="blocks.html#equations"><i class="fa fa-check"></i><b>5.1</b> Equations</a></li>
<li class="chapter" data-level="5.2" data-path="blocks.html"><a href="blocks.html#theorems-and-proofs"><i class="fa fa-check"></i><b>5.2</b> Theorems and proofs</a></li>
<li class="chapter" data-level="5.3" data-path="blocks.html"><a href="blocks.html#callout-blocks"><i class="fa fa-check"></i><b>5.3</b> Callout blocks</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="sharing-your-book.html"><a href="sharing-your-book.html"><i class="fa fa-check"></i><b>6</b> Sharing your book</a>
<ul>
<li class="chapter" data-level="6.1" data-path="sharing-your-book.html"><a href="sharing-your-book.html#publishing"><i class="fa fa-check"></i><b>6.1</b> Publishing</a></li>
<li class="chapter" data-level="6.2" data-path="sharing-your-book.html"><a href="sharing-your-book.html#pages"><i class="fa fa-check"></i><b>6.2</b> 404 pages</a></li>
<li class="chapter" data-level="6.3" data-path="sharing-your-book.html"><a href="sharing-your-book.html#metadata-for-sharing"><i class="fa fa-check"></i><b>6.3</b> Metadata for sharing</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">scMINER: a mutual information-based framework for identifying hidden drivers from single-cell omics data</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="data-intake" class="section level1 hasAnchor" number="3">
<h1><span class="header-section-number">Chapter 3</span> Data intake<a href="data-intake.html#data-intake" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>In this chapter, we will introduce you how to create the SparseExpressionSet(SparseEset) objects from inputs of multiple formats.</p>
<div id="genereate-gene-expression-matrix" class="section level2 hasAnchor" number="3.1">
<h2><span class="header-section-number">3.1</span> Genereate gene expression matrix<a href="data-intake.html#genereate-gene-expression-matrix" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>scMINER provides portable functions to generate the <strong>gene expression matrix</strong> from 4 most common file formats of scRNA-seq data, including <strong>sparse matrix by 10x Genomics</strong>, <strong>text-table file</strong>, <strong>HDF5 file by 10x Genomics</strong> and <strong>H5AD file</strong>.</p>
<div id="from-data-directory-by-10x-genomics" class="section level3 hasAnchor" number="3.1.1">
<h3><span class="header-section-number">3.1.1</span> From data directory by 10x Genomics<a href="data-intake.html#from-data-directory-by-10x-genomics" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>This is the most popular input format of scRNA-seq data generated by 10x Genomics. Usually, the data directory contains three files:</p>
<ul>
<li><strong>matrix.mtx</strong>: a sparse matrix format containing the raw UMI count per cell-gene combination</li>
<li><strong>barcodes.tsv</strong>: a tab-separated matrix containing the cell barcodes</li>
<li><strong>features.tsv</strong>: a tab-separated matrix containing the features/genes and their annotations</li>
</ul>
<p>For more details about this format, please check out <a href="https://www.10xgenomics.com/support/software/cell-ranger/latest/analysis/outputs/cr-outputs-mex-matrices">here</a>.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="data-intake.html#cb9-1" tabindex="-1"></a>data_dir <span class="ot">&lt;-</span> <span class="fu">system.file</span>(<span class="st">&quot;extdata/demo_inputs/cell_matrix_10x&quot;</span>, <span class="at">package =</span> <span class="st">&quot;scMINER&quot;</span>)</span>
<span id="cb9-2"><a href="data-intake.html#cb9-2" tabindex="-1"></a><span class="fu">list.files</span>(<span class="at">path =</span> data_dir, <span class="at">full.names =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<pre><code>## [1] &quot;barcodes.tsv.gz&quot; &quot;features.tsv.gz&quot; &quot;matrix.mtx.gz&quot;</code></pre>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="data-intake.html#cb11-1" tabindex="-1"></a>demo1_mtx <span class="ot">&lt;-</span> <span class="fu">readInput_10x.dir</span>(<span class="at">input_dir =</span> data_dir, <span class="at">featureType =</span> <span class="st">&quot;gene_symbol&quot;</span>, <span class="at">removeSuffix =</span> <span class="cn">TRUE</span>, <span class="at">addPrefix =</span> <span class="st">&quot;demo1&quot;</span>)</span></code></pre></div>
<pre><code>## Reading 10x Genomcis data from: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/library/scMINER/extdata/demo_inputs/cell_matrix_10x ...
##  Multiple data modalities were found: Gene Expression, Peaks . Only the gene expression data (under &quot;Gene Expression&quot;) was kept.
## Done! The sparse gene expression matrix has been generated: 500 genes, 100 cells.</code></pre>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="data-intake.html#cb13-1" tabindex="-1"></a>demo1_mtx[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span></code></pre></div>
<pre><code>## 5 x 5 sparse Matrix of class &quot;dgTMatrix&quot;
##            demo1_AAACAGCCAAACGGGC demo1_AAACAGCCAACTAGCC demo1_AAACAGCCAATTAGGA
## AL590822.3                      .                      .                      .
## MORN1                           .                      .                      .
## AL589739.1                      .                      .                      .
## AL513477.2                      .                      .                      .
## RER1                            .                      .                      .
##            demo1_AAACAGCCAGCCAGTT demo1_AAACATGCAAAGCTCC
## AL590822.3                      .                      .
## MORN1                           .                      .
## AL589739.1                      .                      .
## AL513477.2                      .                      .
## RER1                            1                      .</code></pre>
<p>The <code>readInput_10x.dir()</code> function can handle these conditions:</p>
<ul>
<li>Alternative file names for feature data: for the datasets generated by CellRanger 3.0 or earlier, the file name is <strong>genes.tsv</strong>;</li>
<li>Compressed input files: one or more input files are compressed, usually in “<strong>.gz</strong>” format;</li>
<li>Data with multiple modalities: like the single cell multiome data. In this case, <code>readInput_10x.dir()</code> only retains the data of “Gene Expression” by default.</li>
</ul>
</div>
<div id="from-text-table-file" class="section level3 hasAnchor" number="3.1.2">
<h3><span class="header-section-number">3.1.2</span> From text-table file<a href="data-intake.html#from-text-table-file" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>This is definitely the most compatible text format for scRNA-seq data. which can be used by all single-cell RNA-seq technologies, like 10x Genomics, Smart-Seq, Drop-Seq and more. The commonly used text table file formats include <strong>txt</strong> (text file format), <strong>csv</strong> (comma-separated values) and <strong>tsv</strong> (tab-separated values).</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="data-intake.html#cb15-1" tabindex="-1"></a>table_file <span class="ot">&lt;-</span> <span class="fu">system.file</span>(<span class="st">&quot;extdata/demo_inputs/table_file/demoData3.txt.gz&quot;</span>, <span class="at">package =</span> <span class="st">&quot;scMINER&quot;</span>)</span>
<span id="cb15-2"><a href="data-intake.html#cb15-2" tabindex="-1"></a>demo3_mtx <span class="ot">&lt;-</span> <span class="fu">readInput_table</span>(<span class="at">table_file =</span> table_file, <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">is.geneBYcell =</span> <span class="cn">TRUE</span>, <span class="at">removeSuffix =</span> <span class="cn">TRUE</span>, <span class="at">addPrefix =</span> <span class="st">&quot;demo3&quot;</span>) <span class="co"># set is.geneBYcell = FALSE to read features in columns and cell in in rows</span></span></code></pre></div>
<p><strong><em>NOTE:</em></strong> A major concern to read the gene expression matrix from text-table files is that <strong>the special characters in column names might change to dots (“.”)</strong>, especially when the matrix is organized in cells by genes. This may cause failures in the identification of mitochondrial genes (usually defined by “MT-|mt-”) or spike-in RNAs (usually defined by “ERCC-|Ercc-”). The <code>readInput_table()</code> function has set <code>check.names = FALSE</code> to avoid this issue. However, if this issue already exists in the source data, you will have to fix it manually.</p>
</div>
<div id="from-hdf5-file-by-10x-genomics" class="section level3 hasAnchor" number="3.1.3">
<h3><span class="header-section-number">3.1.3</span> From HDF5 file by 10x Genomics<a href="data-intake.html#from-hdf5-file-by-10x-genomics" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>This is another popular input format of scRNA-seq data generated by 10x Genomics. The Hierarchical Data Format version 5 (HDF5 or H5) is a binary format that can compress and access data much more efficiently than text formats. It’s super useful when dealing with large datasets.</p>
<p>For more details about this format, please check out <a href="https://www.10xgenomics.com/support/software/cell-ranger/latest/analysis/outputs/cr-outputs-h5-matrices">here</a>.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="data-intake.html#cb16-1" tabindex="-1"></a>h5_file <span class="ot">&lt;-</span> <span class="fu">system.file</span>(<span class="st">&quot;extdata/demo_inputs/hdf5_10x/demoData2.h5&quot;</span>, <span class="at">package =</span> <span class="st">&quot;scMINER&quot;</span>)</span>
<span id="cb16-2"><a href="data-intake.html#cb16-2" tabindex="-1"></a>demo2_mtx <span class="ot">&lt;-</span> <span class="fu">readInput_10x.h5</span>(<span class="at">h5_file =</span> h5_file, <span class="at">featureType =</span> <span class="st">&quot;gene_symbol&quot;</span>, <span class="at">removeSuffix =</span> <span class="cn">TRUE</span>, <span class="at">addPrefix =</span> <span class="st">&quot;demo2&quot;</span>)</span></code></pre></div>
<p><strong><em>NOTE:</em></strong> The <code>readInput_10x.h5()</code> function is developed exclusively for the HDF5 file generated by CellRanger of 10x Genomics. The HDF5 files from other source may have different hierarchical structures and can not be read by this function.</p>
</div>
<div id="from-h5ad-file" class="section level3 hasAnchor" number="3.1.4">
<h3><span class="header-section-number">3.1.4</span> From H5AD file<a href="data-intake.html#from-h5ad-file" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The H5AD file is another well-used format for scRNA-seq data. Derived from HDF5 file format, the H5AD file is designed to store large amounts of data efficiently and to facilitate fast access to subsets of the data. Now it’s getting more and more popular in scRNA-seq data analysis, visualization and sharing.</p>
<p>For more details about this format, please check out <a href="https://anndata.dynverse.org/">here</a>.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="data-intake.html#cb17-1" tabindex="-1"></a>h5ad_file <span class="ot">&lt;-</span> <span class="fu">system.file</span>(<span class="st">&quot;extdata/demo_inputs/h5ad_file/demoData4.h5ad&quot;</span>, <span class="at">package =</span> <span class="st">&quot;scMINER&quot;</span>)</span>
<span id="cb17-2"><a href="data-intake.html#cb17-2" tabindex="-1"></a>demo4_obj <span class="ot">&lt;-</span> <span class="fu">readInput_h5ad</span>(<span class="at">h5ad_file =</span> h5ad_file, <span class="at">removeSuffix =</span> <span class="cn">TRUE</span>, <span class="at">addPrefix =</span> <span class="st">&quot;demo4&quot;</span>) <span class="co"># set is.geneBYcell = FALSE to read features in columns and cell in in rows</span></span></code></pre></div>
<p><strong><em>NOTE:</em></strong> Unlike the other three readInput functions which return a gene expression matrx, the <code>readInput_h5ad()</code> returns an AnnData object. Here are the key components of an AnnData object:</p>
<ul>
<li><strong>X</strong>: the primary data matrix, cells by genes;</li>
<li><strong>obs</strong>: observations (cells) metadata;</li>
<li><strong>var</strong>: variables (features/genes) metadata.</li>
</ul>
</div>
</div>
<div id="create-sparseeset-object" class="section level2 hasAnchor" number="3.2">
<h2><span class="header-section-number">3.2</span> Create SparseEset object<a href="data-intake.html#create-sparseeset-object" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="solely-from-the-gene-expression-matrix" class="section level3 hasAnchor" number="3.2.1">
<h3><span class="header-section-number">3.2.1</span> Solely from the gene expression matrix<a href="data-intake.html#solely-from-the-gene-expression-matrix" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>This is <strong>the most commonly used way</strong> to create the sparse eSet object with scMINER:</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="data-intake.html#cb18-1" tabindex="-1"></a>pbmc14k_raw.eset <span class="ot">&lt;-</span> <span class="fu">createSparseEset</span>(<span class="at">input_matrix =</span> pbmc14k_rawCount, <span class="at">projectID =</span> <span class="st">&quot;PBMC14k&quot;</span>, <span class="at">addMetaData =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<pre><code>## Creating sparse eset from the input_matrix ...
##  Adding meta data based on input_matrix ...
## Done! The sparse eset has been generated: 17986 genes, 14000 cells.</code></pre>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="data-intake.html#cb20-1" tabindex="-1"></a>pbmc14k_raw.eset</span></code></pre></div>
<pre><code>## SparseExpressionSet (storageMode: environment)
## assayData: 17986 features, 14000 samples 
##   element names: exprs 
## protocolData: none
## phenoData
##   sampleNames: CACTTTGACGCAAT GTTACGGAAACGAA ... ACGTGCCTTAAAGG (14000
##     total)
##   varLabels: CellID projectID ... pctSpikeIn (6 total)
##   varMetadata: labelDescription
## featureData
##   featureNames: AL627309.1 AP006222.2 ... SRSF10.1 (17986 total)
##   fvarLabels: GeneSymbol nCell
##   fvarMetadata: labelDescription
## experimentData: use &#39;experimentData(object)&#39;
## Annotation:</code></pre>
<ul>
<li><strong>input_matrix</strong>: it’s usually but not limited to a sparse matrix of raw UMI count.
<ul>
<li>As for the data format, it accepts <strong><code>dgCMatrix</code></strong>, <strong><code>dgTMatrix</code></strong>, <strong><code>dgeMatrix</code></strong>, <strong><code>matrix</code></strong>, <strong><code>data.frame</code></strong>.</li>
<li>As for the type of quantification measures, it takes <strong>raw counts</strong>, normalized counts (e.g. <strong><code>CPM</code></strong> or <strong><code>CP10k</code></strong>), <strong><code>TPM</code></strong> (Transcripts Per Million), <strong><code>FPKM/RPKM</code></strong> (Fragments/Reads Per Kilobase of transcript per Million) and others.</li>
<li><strong>What if a data frame object is given to it?</strong> When a non-matrix table is passed to <code>input_matrix</code> argument, the <code>createSparseEset()</code> function will automatically convert it to a matrix. And it the matrix, either converted from other format or directly passed from users, is not sparse. <code>createSparseEset()</code> will automatically convert it into sparse matrix, by default. This is controlled by another argument called <code>do.sparseConversion</code>, the default of which is <code>TRUE</code>. It’s not recommended but the users can set it as <code>FALSE</code> to disable the conversion. Then <code>createSparseEset()</code> will create the eSet based on the regular matrix.</li>
</ul></li>
<li><strong>addMetaData</strong>: when this argument is set <code>TRUE</code> (this is the default), <code>createSparseEset()</code> will automatically generate 5 statistics, 4 for cells and 1 for features, and add them into the <code>phenoData</code> and <code>featureData</code> slots. These 5 statistics will be used in quality control and data filtration.</li>
</ul>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="data-intake.html#cb22-1" tabindex="-1"></a><span class="do">## check the phenoData: metadata of cells</span></span>
<span id="cb22-2"><a href="data-intake.html#cb22-2" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">pData</span>(pbmc14k_raw.eset))</span></code></pre></div>
<pre><code>##                        CellID projectID nUMI nFeature    pctMito pctSpikeIn
## CACTTTGACGCAAT CACTTTGACGCAAT   PBMC14k  764      354 0.01832461          0
## GTTACGGAAACGAA GTTACGGAAACGAA   PBMC14k  956      442 0.01569038          0
## AGTCACGACAGGAG AGTCACGACAGGAG   PBMC14k 7940     2163 0.01977330          0
## TTCGAGGACCAGTA TTCGAGGACCAGTA   PBMC14k 4177     1277 0.01149150          0
## CACTTATGAGTCGT CACTTATGAGTCGT   PBMC14k  629      323 0.02066773          0
## GCATGTGATTCTGT GCATGTGATTCTGT   PBMC14k  875      427 0.02628571          0</code></pre>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="data-intake.html#cb24-1" tabindex="-1"></a><span class="do">## check the featureData: metadata of features</span></span>
<span id="cb24-2"><a href="data-intake.html#cb24-2" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">fData</span>(pbmc14k_raw.eset))</span></code></pre></div>
<pre><code>##                  GeneSymbol nCell
## AL627309.1       AL627309.1    50
## AP006222.2       AP006222.2     2
## RP11-206L10.3 RP11-206L10.3     1
## RP11-206L10.2 RP11-206L10.2    33
## RP11-206L10.9 RP11-206L10.9    17
## LINC00115         LINC00115   115</code></pre>
</div>
<div id="using-self-customized-meta-data" class="section level3 hasAnchor" number="3.2.2">
<h3><span class="header-section-number">3.2.2</span> Using self-customized meta data<a href="data-intake.html#using-self-customized-meta-data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>In some cases, you may have more meta data of either cells (e.g. sample id, treatment condition) or features (e.g. gene full name, gene type, genome location) which will be used in downstream analysis and you do want to add them into the sparse eSet object. the <code>createSparseEset()</code> function provides another two arguments, <strong><code>cellData</code></strong> and <strong><code>featureData</code></strong>, to take the self-customized meta data. For the PBMC14k dataset, we have the true labels of cell type and would like to add them to the sparse eSet object.</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="data-intake.html#cb26-1" tabindex="-1"></a><span class="do">## read the true labels of cell type for PBMC14k dataset</span></span>
<span id="cb26-2"><a href="data-intake.html#cb26-2" tabindex="-1"></a>true_label <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="fu">system.file</span>(<span class="st">&quot;extdata/demo_pbmc14k/PBMC14k_trueLabel.txt.gz&quot;</span>, <span class="at">package =</span> <span class="st">&quot;scMINER&quot;</span>), <span class="at">header =</span> T, <span class="at">row.names =</span> <span class="dv">1</span>, <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">quote =</span> <span class="st">&quot;&quot;</span>, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb26-3"><a href="data-intake.html#cb26-3" tabindex="-1"></a></span>
<span id="cb26-4"><a href="data-intake.html#cb26-4" tabindex="-1"></a><span class="fu">head</span>(true_label)</span></code></pre></div>
<pre><code>##                trueLabel_full trueLabel
## CACTTTGACGCAAT CD14+ Monocyte  Monocyte
## GTTACGGAAACGAA CD14+ Monocyte  Monocyte
## AGTCACGACAGGAG CD14+ Monocyte  Monocyte
## TTCGAGGACCAGTA CD14+ Monocyte  Monocyte
## CACTTATGAGTCGT CD14+ Monocyte  Monocyte
## GCATGTGATTCTGT CD14+ Monocyte  Monocyte</code></pre>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="data-intake.html#cb28-1" tabindex="-1"></a><span class="fu">table</span>(true_label<span class="sc">$</span>trueLabel_full)</span></code></pre></div>
<pre><code>## 
##               CD14+ Monocyte                      CD19+ B 
##                         2000                         2000 
##              CD4+/CD25 T Reg   CD4+/CD45RA+/CD25- Naive T 
##                         2000                         2000 
##          CD4+/CD45RO+ Memory                     CD56+ NK 
##                         2000                         2000 
## CD8+/CD45RA+ Naive Cytotoxic 
##                         2000</code></pre>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="data-intake.html#cb30-1" tabindex="-1"></a><span class="do">## the true_label much cover all cells in the expression matrix</span></span>
<span id="cb30-2"><a href="data-intake.html#cb30-2" tabindex="-1"></a><span class="fu">table</span>(<span class="fu">colnames</span>(pbmc14k_rawCount) <span class="sc">%in%</span> <span class="fu">row.names</span>(true_label))</span></code></pre></div>
<pre><code>## 
##  TRUE 
## 14000</code></pre>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="data-intake.html#cb32-1" tabindex="-1"></a><span class="do">## create the sparse eSet object using the true_label</span></span>
<span id="cb32-2"><a href="data-intake.html#cb32-2" tabindex="-1"></a>pbmc14k_raw.eset <span class="ot">&lt;-</span> <span class="fu">createSparseEset</span>(<span class="at">input_matrix =</span> pbmc14k_rawCount, <span class="at">cellData =</span> true_label, <span class="at">featureData =</span> <span class="cn">NULL</span>, <span class="at">projectID =</span> <span class="st">&quot;PBMC14k&quot;</span>, <span class="at">addMetaData =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<pre><code>## Creating sparse eset from the input_matrix ...
##  Adding meta data based on input_matrix ...
## Done! The sparse eset has been generated: 17986 genes, 14000 cells.</code></pre>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="data-intake.html#cb34-1" tabindex="-1"></a><span class="do">## check the true labels of cell type from sparse eSet object</span></span>
<span id="cb34-2"><a href="data-intake.html#cb34-2" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">pData</span>(pbmc14k_raw.eset))</span></code></pre></div>
<pre><code>##                trueLabel_full trueLabel projectID nUMI nFeature    pctMito
## CACTTTGACGCAAT CD14+ Monocyte  Monocyte   PBMC14k  764      354 0.01832461
## GTTACGGAAACGAA CD14+ Monocyte  Monocyte   PBMC14k  956      442 0.01569038
## AGTCACGACAGGAG CD14+ Monocyte  Monocyte   PBMC14k 7940     2163 0.01977330
## TTCGAGGACCAGTA CD14+ Monocyte  Monocyte   PBMC14k 4177     1277 0.01149150
## CACTTATGAGTCGT CD14+ Monocyte  Monocyte   PBMC14k  629      323 0.02066773
## GCATGTGATTCTGT CD14+ Monocyte  Monocyte   PBMC14k  875      427 0.02628571
##                pctSpikeIn         CellID
## CACTTTGACGCAAT          0 CACTTTGACGCAAT
## GTTACGGAAACGAA          0 GTTACGGAAACGAA
## AGTCACGACAGGAG          0 AGTCACGACAGGAG
## TTCGAGGACCAGTA          0 TTCGAGGACCAGTA
## CACTTATGAGTCGT          0 CACTTATGAGTCGT
## GCATGTGATTCTGT          0 GCATGTGATTCTGT</code></pre>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="data-intake.html#cb36-1" tabindex="-1"></a><span class="fu">table</span>(<span class="fu">pData</span>(pbmc14k_raw.eset)<span class="sc">$</span>trueLabel_full)</span></code></pre></div>
<pre><code>## 
##               CD14+ Monocyte                      CD19+ B 
##                         2000                         2000 
##              CD4+/CD25 T Reg   CD4+/CD45RA+/CD25- Naive T 
##                         2000                         2000 
##          CD4+/CD45RO+ Memory                     CD56+ NK 
##                         2000                         2000 
## CD8+/CD45RA+ Naive Cytotoxic 
##                         2000</code></pre>
</div>
<div id="from-multiple-samples" class="section level3 hasAnchor" number="3.2.3">
<h3><span class="header-section-number">3.2.3</span> From multiple samples<a href="data-intake.html#from-multiple-samples" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>What if you have multiple samples for one project?</strong> Now it’s pretty common to profile multiple samples of the same tissue but under different conditions (e.g. drug treatment) in one project. Analyzing these samples one by one is crucial, and analyzing them in a combined manner may give you more prospects. For this purpose, scMINER provides a function, <code>combineSparseEset()</code>, to easily combine the sparse eSet objects of multiple samples.</p>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="data-intake.html#cb38-1" tabindex="-1"></a><span class="do">## create a sparse eSet object of each sample to combined</span></span>
<span id="cb38-2"><a href="data-intake.html#cb38-2" tabindex="-1"></a>demo1_mtx <span class="ot">&lt;-</span> <span class="fu">readInput_10x.dir</span>(<span class="at">input_dir =</span> <span class="fu">system.file</span>(<span class="st">&quot;extdata/demo_inputs/cell_matrix_10x&quot;</span>, <span class="at">package =</span> <span class="st">&quot;scMINER&quot;</span>), <span class="at">featureType =</span> <span class="st">&quot;gene_symbol&quot;</span>, <span class="at">removeSuffix =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<pre><code>## Reading 10x Genomcis data from: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/library/scMINER/extdata/demo_inputs/cell_matrix_10x ...
##  Multiple data modalities were found: Gene Expression, Peaks . Only the gene expression data (under &quot;Gene Expression&quot;) was kept.
## Done! The sparse gene expression matrix has been generated: 500 genes, 100 cells.</code></pre>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="data-intake.html#cb40-1" tabindex="-1"></a>demo1.eset <span class="ot">&lt;-</span> <span class="fu">createSparseEset</span>(<span class="at">input_matrix =</span> demo1_mtx, <span class="at">projectID =</span> <span class="st">&quot;demo1&quot;</span>, <span class="at">addMetaData =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<pre><code>## Creating sparse eset from the input_matrix ...
##  Adding meta data based on input_matrix ...
## Done! The sparse eset has been generated: 500 genes, 100 cells.</code></pre>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="data-intake.html#cb42-1" tabindex="-1"></a>demo2_mtx <span class="ot">&lt;-</span> <span class="fu">readInput_table</span>(<span class="at">table_file =</span> <span class="fu">system.file</span>(<span class="st">&quot;extdata/demo_inputs/table_file/demoData3.txt.gz&quot;</span>, <span class="at">package =</span> <span class="st">&quot;scMINER&quot;</span>), <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">is.geneBYcell =</span> <span class="cn">TRUE</span>, <span class="at">removeSuffix =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<pre><code>## Reading table file: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/library/scMINER/extdata/demo_inputs/table_file/demoData3.txt.gz ...
##  Suffix removal was specified but skipped, since some barcodes do not carry &quot;-1&quot; suffix.
## Done! The sparse gene expression matrix has been generated: 1000 genes, 100 cells.</code></pre>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="data-intake.html#cb44-1" tabindex="-1"></a>demo2.eset <span class="ot">&lt;-</span> <span class="fu">createSparseEset</span>(<span class="at">input_matrix =</span> demo2_mtx, <span class="at">projectID =</span> <span class="st">&quot;demo2&quot;</span>, <span class="at">addMetaData =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<pre><code>## Creating sparse eset from the input_matrix ...
##  Adding meta data based on input_matrix ...
## Done! The sparse eset has been generated: 1000 genes, 100 cells.</code></pre>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="data-intake.html#cb46-1" tabindex="-1"></a><span class="do">## combine the 4 sparse eSet objects</span></span>
<span id="cb46-2"><a href="data-intake.html#cb46-2" tabindex="-1"></a>combined.eset <span class="ot">&lt;-</span> <span class="fu">combineSparseEset</span>(<span class="at">eset_list =</span> <span class="fu">c</span>(demo1.eset, demo2.eset),</span>
<span id="cb46-3"><a href="data-intake.html#cb46-3" tabindex="-1"></a>                                   <span class="at">projectID =</span> <span class="fu">c</span>(<span class="st">&quot;sample1&quot;</span>, <span class="st">&quot;sample2&quot;</span>),</span>
<span id="cb46-4"><a href="data-intake.html#cb46-4" tabindex="-1"></a>                                   <span class="at">addPrefix =</span> <span class="fu">c</span>(<span class="st">&quot;demo1&quot;</span>, <span class="st">&quot;demo2&quot;</span>),</span>
<span id="cb46-5"><a href="data-intake.html#cb46-5" tabindex="-1"></a>                                   <span class="at">addSurfix =</span> <span class="cn">NULL</span>, <span class="at">addMetaData =</span> <span class="cn">TRUE</span>, <span class="at">imputeNA =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<pre><code>## Combining the input sparse eSets ...
## NA values were found in the merged matrix and have been replaced by the minimum value:  0 .
## Adding meta data based on merged data matrix ...
## Done! The combined sparse eset has been generated: 1500 genes, 200 cells.</code></pre>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb48-1"><a href="data-intake.html#cb48-1" tabindex="-1"></a><span class="fu">dim</span>(combined.eset)</span></code></pre></div>
<pre><code>## Features  Samples 
##     1500      200</code></pre>
<p>A few questions you may have about the <code>combineSparseEset()</code> function:
- <strong>What if the input eSets have different features?</strong> <code>combineSparseEset()</code> ALWAYS keep all features from the input eSets, and generate NA values wherever the data is not available. By default, this function impute the NA values with the minimum value of the combined matrix, which is usually but not always zero. If this imputation method doesn’t fit your study, you can set <code>imputeNA</code> to <code>FALSE</code> to disable it. If so, the NAs will retain in the eSet object, and you can manually impute them with your own method.
- <strong>What if the input eSets have some same cell barcodes?</strong> <code>combineSparseEset()</code> ALWAYS keep all cells from the input eSets, and will report an error when same barcodes are found in different input eSets. This function provides two arguments, <code>addPrefix</code> and <code>addSurfix</code>, to solve this issue. You can easily avoid the same barcodes of different input eSets by adding a eSet-specific prefix and/or surfix to the barcodes.</p>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb50-1"><a href="data-intake.html#cb50-1" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">pData</span>(combined.eset))</span></code></pre></div>
<pre><code>##                                        CellID projectID nUMI nFeature pctMito
## demo1_AAACAGCCAAACGGGC demo1_AAACAGCCAAACGGGC   sample1  119       43       0
## demo1_AAACAGCCAACTAGCC demo1_AAACAGCCAACTAGCC   sample1   55       28       0
## demo1_AAACAGCCAATTAGGA demo1_AAACAGCCAATTAGGA   sample1   45       20       0
## demo1_AAACAGCCAGCCAGTT demo1_AAACAGCCAGCCAGTT   sample1  175       44       0
## demo1_AAACATGCAAAGCTCC demo1_AAACATGCAAAGCTCC   sample1   51       31       0
## demo1_AAACATGCAATAGCCC demo1_AAACATGCAATAGCCC   sample1  121       44       0
##                        pctSpikeIn
## demo1_AAACAGCCAAACGGGC          0
## demo1_AAACAGCCAACTAGCC          0
## demo1_AAACAGCCAATTAGGA          0
## demo1_AAACAGCCAGCCAGTT          0
## demo1_AAACATGCAAAGCTCC          0
## demo1_AAACATGCAATAGCCC          0</code></pre>
<ul>
<li><strong>I have some customized column in the phenoData and/or featureData slots. How does <code>combineSparseEset()</code> handle them?</strong> <code>combineSparseEset()</code> only keep the columns of phenoData and featureData that shared by all input eSets. Your customized columns would be kept only when they are available in all input eSets.</li>
<li><strong>Are the 5 meta data statistics in the combined eSet still same with those generated in each eSet?</strong> No. By default, <code>combineSparseEset()</code> will update (add, if they are not available in input eSets) these 5 meta data statistics based on the combined matrix. It’s not recommended but you can disable it by setting <code>addMataData</code> to <code>FALSE</code>.</li>
</ul>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="get-started.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="footnotes-and-citations.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/03-data-intake.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
