<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>8.1 Supervised cell type annotation | scMINER: a mutual information-based framework for identifying hidden drivers from single-cell omics data</title>
  <meta name="description" content="A tutorial of scMINER" />
  <meta name="generator" content="bookdown 0.40 and GitBook 2.6.7" />

  <meta property="og:title" content="8.1 Supervised cell type annotation | scMINER: a mutual information-based framework for identifying hidden drivers from single-cell omics data" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="A tutorial of scMINER" />
  <meta name="github-repo" content="QingfeiPan/scMINER_qpan" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="8.1 Supervised cell type annotation | scMINER: a mutual information-based framework for identifying hidden drivers from single-cell omics data" />
  
  <meta name="twitter:description" content="A tutorial of scMINER" />
  

<meta name="author" content="Qingfei Pan" />


<meta name="date" content="2024-07-31" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="cell-type-annotation.html"/>
<link rel="next" href="unsupervised-cell-type-annotation.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A tutorial of scMINER</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><img src="images/scMINER_logo.png" alt="" width="250px"></a></li>
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="a-few-concepts.html"><a href="a-few-concepts.html"><i class="fa fa-check"></i><b>1.1</b> A few concepts</a>
<ul>
<li class="chapter" data-level="" data-path="a-few-concepts.html"><a href="a-few-concepts.html#sparseeset"><i class="fa fa-check"></i>SparseEset</a></li>
<li class="chapter" data-level="" data-path="a-few-concepts.html"><a href="a-few-concepts.html#mutual-information"><i class="fa fa-check"></i>Mutual Information</a></li>
<li class="chapter" data-level="" data-path="a-few-concepts.html"><a href="a-few-concepts.html#gene-activity"><i class="fa fa-check"></i>Gene Activity</a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="why-use-scminer.html"><a href="why-use-scminer.html"><i class="fa fa-check"></i><b>1.2</b> Why use scMINER</a></li>
<li class="chapter" data-level="1.3" data-path="citation.html"><a href="citation.html"><i class="fa fa-check"></i><b>1.3</b> Citation</a></li>
<li class="chapter" data-level="1.4" data-path="support.html"><a href="support.html"><i class="fa fa-check"></i><b>1.4</b> Support</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="get-started.html"><a href="get-started.html"><i class="fa fa-check"></i><b>2</b> Get started</a>
<ul>
<li class="chapter" data-level="2.1" data-path="installation.html"><a href="installation.html"><i class="fa fa-check"></i><b>2.1</b> Installation</a>
<ul>
<li class="chapter" data-level="" data-path="installation.html"><a href="installation.html#install-scminer-r-package"><i class="fa fa-check"></i>Install scMINER R package</a></li>
<li class="chapter" data-level="" data-path="installation.html"><a href="installation.html#install-mica-and-sjaracne"><i class="fa fa-check"></i>Install MICA and SJARACNe</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="demo-data.html"><a href="demo-data.html"><i class="fa fa-check"></i><b>2.2</b> Demo data</a></li>
<li class="chapter" data-level="2.3" data-path="create-project-space.html"><a href="create-project-space.html"><i class="fa fa-check"></i><b>2.3</b> Create project space</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="generate-gene-expresion-matrix.html"><a href="generate-gene-expresion-matrix.html"><i class="fa fa-check"></i><b>3</b> Generate gene expresion matrix</a>
<ul>
<li class="chapter" data-level="3.1" data-path="from-data-directory-by-10x-genomics.html"><a href="from-data-directory-by-10x-genomics.html"><i class="fa fa-check"></i><b>3.1</b> From data directory by 10x Genomics</a></li>
<li class="chapter" data-level="3.2" data-path="from-text-table-file.html"><a href="from-text-table-file.html"><i class="fa fa-check"></i><b>3.2</b> From text-table file</a></li>
<li class="chapter" data-level="3.3" data-path="from-hdf5-file-by-10x-genomics.html"><a href="from-hdf5-file-by-10x-genomics.html"><i class="fa fa-check"></i><b>3.3</b> From HDF5 file by 10x Genomics</a></li>
<li class="chapter" data-level="3.4" data-path="from-h5ad-file.html"><a href="from-h5ad-file.html"><i class="fa fa-check"></i><b>3.4</b> From H5AD file</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="create-sparseeset-object.html"><a href="create-sparseeset-object.html"><i class="fa fa-check"></i><b>4</b> Create SparseEset object</a>
<ul>
<li class="chapter" data-level="4.1" data-path="solely-from-the-gene-expression-matrix.html"><a href="solely-from-the-gene-expression-matrix.html"><i class="fa fa-check"></i><b>4.1</b> Solely from the gene expression matrix</a></li>
<li class="chapter" data-level="4.2" data-path="using-self-customized-meta-data.html"><a href="using-self-customized-meta-data.html"><i class="fa fa-check"></i><b>4.2</b> Using self-customized meta data</a></li>
<li class="chapter" data-level="4.3" data-path="from-multiple-samples.html"><a href="from-multiple-samples.html"><i class="fa fa-check"></i><b>4.3</b> From multiple samples</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="data-filtration.html"><a href="data-filtration.html"><i class="fa fa-check"></i><b>5</b> Data filtration</a>
<ul>
<li class="chapter" data-level="5.1" data-path="qc-report.html"><a href="qc-report.html"><i class="fa fa-check"></i><b>5.1</b> QC report</a></li>
<li class="chapter" data-level="5.2" data-path="filter-the-sparse-eset-object.html"><a href="filter-the-sparse-eset-object.html"><i class="fa fa-check"></i><b>5.2</b> Filter the sparse eset object</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="filter-the-sparse-eset-object.html"><a href="filter-the-sparse-eset-object.html#data-filtration-with-auto-mode"><i class="fa fa-check"></i><b>5.2.1</b> Data filtration with auto mode</a></li>
<li class="chapter" data-level="5.2.2" data-path="filter-the-sparse-eset-object.html"><a href="filter-the-sparse-eset-object.html#data-filtration-with-manual-mode"><i class="fa fa-check"></i><b>5.2.2</b> Data filtration with manual mode</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="data-normalization.html"><a href="data-normalization.html"><i class="fa fa-check"></i><b>6</b> Data normalization</a></li>
<li class="chapter" data-level="7" data-path="mi-based-clustering-analysis.html"><a href="mi-based-clustering-analysis.html"><i class="fa fa-check"></i><b>7</b> MI-based clustering analysis</a>
<ul>
<li class="chapter" data-level="7.1" data-path="introduction-to-mica.html"><a href="introduction-to-mica.html"><i class="fa fa-check"></i><b>7.1</b> Introduction to MICA</a></li>
<li class="chapter" data-level="7.2" data-path="generate-mica-input.html"><a href="generate-mica-input.html"><i class="fa fa-check"></i><b>7.2</b> Generate MICA input</a></li>
<li class="chapter" data-level="7.3" data-path="run-mica.html"><a href="run-mica.html"><i class="fa fa-check"></i><b>7.3</b> Run MICA</a></li>
<li class="chapter" data-level="7.4" data-path="integrate-mica-outputs-into-sparseeset-object.html"><a href="integrate-mica-outputs-into-sparseeset-object.html"><i class="fa fa-check"></i><b>7.4</b> Integrate MICA outputs into SparseEset object</a></li>
<li class="chapter" data-level="7.5" data-path="visualize-the-mica-output.html"><a href="visualize-the-mica-output.html"><i class="fa fa-check"></i><b>7.5</b> Visualize the MICA output</a>
<ul>
<li class="chapter" data-level="7.5.1" data-path="visualize-the-mica-output.html"><a href="visualize-the-mica-output.html#color-coded-by-cluster-labels"><i class="fa fa-check"></i><b>7.5.1</b> Color-coded by cluster labels</a></li>
<li class="chapter" data-level="7.5.2" data-path="visualize-the-mica-output.html"><a href="visualize-the-mica-output.html#color-coded-by-true-label-of-cell-types"><i class="fa fa-check"></i><b>7.5.2</b> Color-coded by true label of cell types</a></li>
<li class="chapter" data-level="7.5.3" data-path="visualize-the-mica-output.html"><a href="visualize-the-mica-output.html#color-coded-by-numi-for-qc-purpose"><i class="fa fa-check"></i><b>7.5.3</b> Color-coded by nUMI, for QC purpose</a></li>
<li class="chapter" data-level="7.5.4" data-path="visualize-the-mica-output.html"><a href="visualize-the-mica-output.html#color-coded-by-nfeature-for-qc-purpose"><i class="fa fa-check"></i><b>7.5.4</b> Color-coded by nFeature, for QC purpose</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="cell-type-annotation.html"><a href="cell-type-annotation.html"><i class="fa fa-check"></i><b>8</b> Cell type annotation</a>
<ul>
<li class="chapter" data-level="8.1" data-path="supervised-cell-type-annotation.html"><a href="supervised-cell-type-annotation.html"><i class="fa fa-check"></i><b>8.1</b> Supervised cell type annotation</a>
<ul>
<li class="chapter" data-level="8.1.1" data-path="supervised-cell-type-annotation.html"><a href="supervised-cell-type-annotation.html#annotate-using-signature-scores"><i class="fa fa-check"></i><b>8.1.1</b> Annotate using signature scores</a></li>
<li class="chapter" data-level="8.1.2" data-path="supervised-cell-type-annotation.html"><a href="supervised-cell-type-annotation.html#annotate-using-individual-marker-genes"><i class="fa fa-check"></i><b>8.1.2</b> Annotate using individual marker genes</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="unsupervised-cell-type-annotation.html"><a href="unsupervised-cell-type-annotation.html"><i class="fa fa-check"></i><b>8.2</b> Unsupervised cell type annotation</a></li>
<li class="chapter" data-level="8.3" data-path="add-cell-type-annotations-to-sparseexpressionset-object.html"><a href="add-cell-type-annotations-to-sparseexpressionset-object.html"><i class="fa fa-check"></i><b>8.3</b> Add cell type annotations to SparseExpressionSet object</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="network-inference.html"><a href="network-inference.html"><i class="fa fa-check"></i><b>9</b> Network inference</a>
<ul>
<li class="chapter" data-level="9.1" data-path="generate-sjaracne-input-files.html"><a href="generate-sjaracne-input-files.html"><i class="fa fa-check"></i><b>9.1</b> Generate SJARACNe input files</a></li>
<li class="chapter" data-level="9.2" data-path="run-sjaracne.html"><a href="run-sjaracne.html"><i class="fa fa-check"></i><b>9.2</b> Run SJARACNe</a></li>
<li class="chapter" data-level="9.3" data-path="assess-the-quality-of-networks.html"><a href="assess-the-quality-of-networks.html"><i class="fa fa-check"></i><b>9.3</b> Assess the quality of networks</a>
<ul>
<li class="chapter" data-level="9.3.1" data-path="assess-the-quality-of-networks.html"><a href="assess-the-quality-of-networks.html#introduction-to-the-network-file-by-sjaracne"><i class="fa fa-check"></i><b>9.3.1</b> Introduction to the network file by SJARACNe</a></li>
<li class="chapter" data-level="9.3.2" data-path="assess-the-quality-of-networks.html"><a href="assess-the-quality-of-networks.html#generate-network-qc-report"><i class="fa fa-check"></i><b>9.3.2</b> Generate network QC report</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="actvity-based-analysis.html"><a href="actvity-based-analysis.html"><i class="fa fa-check"></i><b>10</b> Actvity-based analysis</a>
<ul>
<li class="chapter" data-level="10.1" data-path="calculate-the-activities.html"><a href="calculate-the-activities.html"><i class="fa fa-check"></i><b>10.1</b> Calculate the activities</a>
<ul>
<li class="chapter" data-level="10.1.1" data-path="calculate-the-activities.html"><a href="calculate-the-activities.html#calculate-activities-per-group"><i class="fa fa-check"></i><b>10.1.1</b> Calculate activities per group</a></li>
<li class="chapter" data-level="10.1.2" data-path="calculate-the-activities.html"><a href="calculate-the-activities.html#calculate-activities-in-batch"><i class="fa fa-check"></i><b>10.1.2</b> Calculate activities in batch</a></li>
<li class="chapter" data-level="10.1.3" data-path="calculate-the-activities.html"><a href="calculate-the-activities.html#save-activity-eset-object"><i class="fa fa-check"></i><b>10.1.3</b> Save activity eset object</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="differential-activity-analysis.html"><a href="differential-activity-analysis.html"><i class="fa fa-check"></i><b>10.2</b> Differential activity analysis</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="session-information.html"><a href="session-information.html"><i class="fa fa-check"></i><b>11</b> Session information</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">scMINER: a mutual information-based framework for identifying hidden drivers from single-cell omics data</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="supervised-cell-type-annotation" class="section level2 hasAnchor" number="8.1">
<h2><span class="header-section-number">8.1</span> Supervised cell type annotation<a href="supervised-cell-type-annotation.html#supervised-cell-type-annotation" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In the past a few years, the scRNA-seq data has dramatically increased in both quality and quantity. For the majority of tissue types, some existing studies on the same/similar tissue type are most likely available, and from these existing studies, we can figure out a list of candidate cell types to expect and curate a list of markers for each of them. In this case, we know the 7 cell types involved in the dataset, and curated a marker list from some existing PBMCs studies.</p>
<div id="annotate-using-signature-scores" class="section level3 hasAnchor" number="8.1.1">
<h3><span class="header-section-number">8.1.1</span> Annotate using signature scores<a href="supervised-cell-type-annotation.html#annotate-using-signature-scores" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Given a marker list of candidate cell types, scMINER can estimate a signature score, which is mathematically <code>the weighted mean of the expression of marker genes involved</code>, for each candidate cell type across all cell cluster. To do so, you will need to generate a signature table with three columns:</p>
<ul>
<li><code>signature_name</code>: name of cell types/signatures;</li>
<li><code>signature_feature</code>: markers genes/features of corresponding cell type/signature;</li>
<li><code>weight</code>: weight of corresponding maker/feature in corresponding cell type/signature. It ranges from -1 to 1, so both positive and negtive markers are supoorted.</li>
</ul>
<div class="sourceCode" id="cb88"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb88-1"><a href="supervised-cell-type-annotation.html#cb88-1" tabindex="-1"></a><span class="do">## Signature table of PBMC14k dataset</span></span>
<span id="cb88-2"><a href="supervised-cell-type-annotation.html#cb88-2" tabindex="-1"></a>signature_table <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="fu">system.file</span>(<span class="st">&quot;extdata/demo_pbmc14k/PBMC14k_signatureTable.txt&quot;</span>, <span class="at">package =</span> <span class="st">&quot;scMINER&quot;</span>), <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">quote =</span> <span class="st">&quot;&quot;</span>, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb88-3"><a href="supervised-cell-type-annotation.html#cb88-3" tabindex="-1"></a><span class="fu">head</span>(signature_table)</span></code></pre></div>
<pre><code>##   signature_name signature_feature weight
## 1       Monocyte              CD14      1
## 2       Monocyte               LYZ      1
## 3       Monocyte            S100A8      1
## 4       Monocyte            S100A9      1
## 5       Monocyte           S100A12      1
## 6             NK            FCGR3A      1</code></pre>
<p>With this signature table, <code>draw_bubbleplot()</code> can estimate the signature scores and visualize them using bubble plot:</p>
<div class="sourceCode" id="cb90"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb90-1"><a href="supervised-cell-type-annotation.html#cb90-1" tabindex="-1"></a><span class="do">## Violin plot of marker genes across clusters</span></span>
<span id="cb90-2"><a href="supervised-cell-type-annotation.html#cb90-2" tabindex="-1"></a><span class="fu">draw_bubbleplot</span>(<span class="at">input_eset =</span> pbmc14k_log2cpm.eset, <span class="at">signature_table =</span> signature_table, <span class="at">group_by =</span> <span class="st">&quot;clusterID&quot;</span>)</span></code></pre></div>
<pre><code>## 31 features of 7 signatures were found in the input eset and will be used in calculation.</code></pre>
<p><img src="_main_files/figure-html/signature-bubble-1.png" width="672" style="display: block; margin: auto;" />
In the bubble plot above, the color of the bubbles is proportional to the mean of signature score, and the size of the bubbles is proportional to the percentage of cells with higher signature score than mean. The cell type of each cluster is clear, except the cluster 7, which shows equally-high signature score of both CD4+ TCM and CD4+ Reg and higher percentage of CD4+ TCM cells.</p>
</div>
<div id="annotate-using-individual-marker-genes" class="section level3 hasAnchor" number="8.1.2">
<h3><span class="header-section-number">8.1.2</span> Annotate using individual marker genes<a href="supervised-cell-type-annotation.html#annotate-using-individual-marker-genes" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>scMINER also provides a variety of functions to visualize the selected features:</p>
<div class="sourceCode" id="cb92"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb92-1"><a href="supervised-cell-type-annotation.html#cb92-1" tabindex="-1"></a><span class="do">## For the demonstration purposes, we picked two well known markers for each of the 7 known cell types, plus &quot;CD3D&quot; and &quot;CD4&quot;.</span></span>
<span id="cb92-2"><a href="supervised-cell-type-annotation.html#cb92-2" tabindex="-1"></a>genes_of_interest <span class="ot">&lt;-</span><span class="fu">c</span>(<span class="st">&quot;CD14&quot;</span>, <span class="st">&quot;LYZ&quot;</span>, <span class="st">&quot;GZMB&quot;</span>, <span class="st">&quot;NKG7&quot;</span>, <span class="st">&quot;CD19&quot;</span>, <span class="st">&quot;MS4A1&quot;</span>, <span class="st">&quot;CD8A&quot;</span>, <span class="st">&quot;CD8B&quot;</span>, <span class="st">&quot;SELL&quot;</span>, <span class="st">&quot;CCR7&quot;</span>, <span class="st">&quot;IL2RA&quot;</span>, <span class="st">&quot;FOXP3&quot;</span>, <span class="st">&quot;IL7R&quot;</span>, <span class="st">&quot;S100A4&quot;</span>, <span class="st">&quot;CD3D&quot;</span>, <span class="st">&quot;CD4&quot;</span>)</span></code></pre></div>
<div id="feature-visualization-violin-plot" class="section level4 hasAnchor" number="8.1.2.1">
<h4><span class="header-section-number">8.1.2.1</span> feature visualization: violin plot<a href="supervised-cell-type-annotation.html#feature-visualization-violin-plot" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb93"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb93-1"><a href="supervised-cell-type-annotation.html#cb93-1" tabindex="-1"></a><span class="do">## Violin plot of marker genes across clusters</span></span>
<span id="cb93-2"><a href="supervised-cell-type-annotation.html#cb93-2" tabindex="-1"></a><span class="fu">feature_vlnplot</span>(<span class="at">input_eset =</span> pbmc14k_log2cpm.eset, <span class="at">features =</span> genes_of_interest, <span class="at">group_by =</span> <span class="st">&quot;clusterID&quot;</span>, <span class="at">ncol =</span> <span class="dv">4</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/featurePlot-vln-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="feature-visualization-box-plot" class="section level4 hasAnchor" number="8.1.2.2">
<h4><span class="header-section-number">8.1.2.2</span> feature visualization: box plot<a href="supervised-cell-type-annotation.html#feature-visualization-box-plot" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb94"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb94-1"><a href="supervised-cell-type-annotation.html#cb94-1" tabindex="-1"></a><span class="do">## Box plot of marker genes across clusters</span></span>
<span id="cb94-2"><a href="supervised-cell-type-annotation.html#cb94-2" tabindex="-1"></a><span class="fu">feature_boxplot</span>(<span class="at">input_eset =</span> pbmc14k_log2cpm.eset, <span class="at">features =</span> genes_of_interest, <span class="at">group_by =</span> <span class="st">&quot;clusterID&quot;</span>, <span class="at">ncol =</span> <span class="dv">4</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/featurePlot-box-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="feature-visualization-scatter-plot" class="section level4 hasAnchor" number="8.1.2.3">
<h4><span class="header-section-number">8.1.2.3</span> feature visualization: scatter plot<a href="supervised-cell-type-annotation.html#feature-visualization-scatter-plot" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb95"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb95-1"><a href="supervised-cell-type-annotation.html#cb95-1" tabindex="-1"></a><span class="do">## UMAP scatter plot of marker genes</span></span>
<span id="cb95-2"><a href="supervised-cell-type-annotation.html#cb95-2" tabindex="-1"></a><span class="fu">feature_scatterplot</span>(<span class="at">input_eset =</span> pbmc14k_log2cpm.eset, <span class="at">features =</span> genes_of_interest, <span class="at">ncol =</span> <span class="dv">4</span>, <span class="at">location_x =</span> <span class="st">&quot;UMAP_1&quot;</span>, <span class="at">location_y =</span>  <span class="st">&quot;UMAP_2&quot;</span>, <span class="at">point.size =</span> <span class="fl">0.5</span>, <span class="at">legend.key_height =</span> <span class="fl">0.3</span>, <span class="at">legend.key_width =</span> <span class="fl">0.2</span>, <span class="at">fontsize.legend_title =</span> <span class="dv">8</span>, <span class="at">fontsize.legend_text =</span> <span class="dv">6</span>, <span class="at">fontsize.axis_title =</span> <span class="dv">8</span>, <span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/featurePlot-scatter-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="feature-visualization-bubble-plot" class="section level4 hasAnchor" number="8.1.2.4">
<h4><span class="header-section-number">8.1.2.4</span> feature visualization: bubble plot<a href="supervised-cell-type-annotation.html#feature-visualization-bubble-plot" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb96"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb96-1"><a href="supervised-cell-type-annotation.html#cb96-1" tabindex="-1"></a><span class="do">## Bubble plot of marker genes across clusters</span></span>
<span id="cb96-2"><a href="supervised-cell-type-annotation.html#cb96-2" tabindex="-1"></a><span class="fu">feature_bubbleplot</span>(<span class="at">input_eset =</span> pbmc14k_log2cpm.eset, <span class="at">features =</span> genes_of_interest, <span class="at">group_by =</span> <span class="st">&quot;clusterID&quot;</span>, <span class="at">xlabel.angle =</span> <span class="dv">45</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/featurePlot-bubble-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="feature-visualization-heatmap" class="section level4 hasAnchor" number="8.1.2.5">
<h4><span class="header-section-number">8.1.2.5</span> feature visualization: heatmap<a href="supervised-cell-type-annotation.html#feature-visualization-heatmap" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb97"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb97-1"><a href="supervised-cell-type-annotation.html#cb97-1" tabindex="-1"></a><span class="do">## Heatmap of marker genes across clusters</span></span>
<span id="cb97-2"><a href="supervised-cell-type-annotation.html#cb97-2" tabindex="-1"></a><span class="fu">feature_heatmap</span>(<span class="at">input_eset =</span> pbmc14k_log2cpm.eset, <span class="at">features =</span> genes_of_interest, <span class="at">group_by =</span> <span class="st">&quot;clusterID&quot;</span>, <span class="at">scale_method =</span> <span class="st">&quot;none&quot;</span>, <span class="at">annotation_columns =</span> <span class="fu">c</span>(<span class="st">&quot;trueLabel&quot;</span>))</span></code></pre></div>
<p><img src="_main_files/figure-html/featurePlot-heatmap-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="cell-type-annotation.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="unsupervised-cell-type-annotation.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/08-celltype-annotation.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
